nomics/src/adapter.ts

```diff
@@ -0,0 +1,47 @@
+import { Requester, Validator, AdapterError } from '@chainlink/external-adapter'
+import { Config, ExecuteWithConfig, ExecuteFactory } from '@chainlink/types'
+import { makeConfig, DEFAULT_ENDPOINT } from './config'
+import { globalmarketcap, price } from './endpoint'
+
+const inputParams = {
+  endpoint: false,
+}
+
+export const execute: ExecuteWithConfig<Config> = async (request, config) => {
+  const validator = new Validator(request, inputParams)
+  if (validator.error) throw validator.error
+
+  Requester.logConfig(config)
+
+  const jobRunID = validator.validated.id
+  const endpoint = validator.validated.data.endpoint || DEFAULT_ENDPOINT
+
+  let data
```

Comment:
You could make a helper to send the `Requester.sucess`, something like:
```js
const _success = (payload: ResponsePayload, result: number) =>
    Requester.success(jobRunID, {
      status: 200,
      data: { sources: [], payload, result },
      result,
    })
 switch (endpoint) {
    case globalmarketcap.Name: {
      const data = await globalmarketcap.execute(config, request)
      return _success(data, data.result)
   ...
}
```

Reply:
I moved the Requester.success inside the specific `endpoint` adapters.

Reply:
You won't need `data` anymore right?